elastic/kibana
# [ResponseOps][Actions] support mustache context variables with periods in the name

Resolves https://github.com/elastic/kibana/issues/127748

## Summary
When we encounter `context` fields with a `.` in the name we expand it to an object and add it back to the context.


### Checklist



### To verify

- Create a rule and add a field with `.` to the action message mustache template. Verify that the message contains the field value when the rule runs.

This is what I did to verify, but you can choose to do what ever is easiest!
- I created a new index by doing the following:
```
PUT _ingest/pipeline/add-timestamp
{
  "processors" : [
    {
      "set" : {
        "field": "@timestamp",
        "value": "{{{_ingest.timestamp}}}"
      }
    }
  ]
}
```
```
POST my-index/_doc/?pipeline=add-timestamp
{
   "my.field": "test"
}
```
I kept adding documents to make sure my rule would hit on something.
- Create an elasticsearch rule with the following query:
<img width="585" alt="Screen Shot 2022-10-20 at 1 36 20 PM" src="https://user-images.githubusercontent.com/109488926/197019732-5a60b810-bd79-4d6f-b653-6541ba6355f3.png">

with a sever log action that had the following message:
```
Elasticsearch query alert is active:
{{#context.hits}}
_source.my.field: {{_source.my.field}}
{{/context.hits}}
```
- Verify that when the rule is active the server log message is populated with the values in `_source.my.field`
